src/gui/guiiconprovider.h

```diff
@@ -30,30 +31,44 @@
 #ifndef GUIICONPROVIDER_H
 #define GUIICONPROVIDER_H
 
+#include "base/bittorrent/torrenthandle.h"
 #include "base/iconprovider.h"
 
 class QIcon;
 
-class GuiIconProvider : public IconProvider
+class GuiIconProvider: public IconProvider
 {
     Q_DISABLE_COPY(GuiIconProvider)
     Q_OBJECT
 
 public:
+
+    enum ThemeFlags
+    {
+        LightTheme = 1,
+        DarkTheme = 2,
+        SmallStatusIcons = 4,
+        LargeStatusIcons = 8,
+    };
+
     static void initInstance();
     static GuiIconProvider *instance();
 
+    int getThemeFlags();
     QIcon getIcon(const QString &iconId);
     QIcon getIcon(const QString &iconId, const QString &fallback);
     QIcon getFlagIcon(const QString &countryIsoCode);
-    QString getIconPath(const QString &iconId);
+    QIcon getStatusIcon(const QString &iconId);
+    QIcon getStatusIcon(const QString &iconId, BitTorrent::TorrentState state);
```

Comment:
BTW, a C++-ish approach to this would be the following:
```C++
struct FuncArgs {
  FuncArgs() {
  // set defaults
  }
  FuncArgs& arg1(Type1 val) {
   m_val1 = val;
   return *this;
  }

  FuncArgs& arg2(Type2 val) {
   m_val2 = val;
   return *this;
  }

private:
    Type1 m_val1;
    Type2 m_val2;
};

void func(const FuncArgs& args = FuncArgs());

func();
func(FuncArgs().arg1(val).arg2(val2));
```